### DISCUSSION TITLE
Can not get Double Jump to work

### DISCUSSION THREAD

**Discussion by jdiperla** [replit/kaboom#145] [Q&A]
I am trying to get the player character to do a dumble jump when the character presses space while the character is already jumping. But it just keeps going:

`keyPress("space", () => {
	// these 2 functions are provided by body() component
	
	
	let jump = 0;
	console.log(jump);
	console.log("space pressed");
	if (player.grounded()) {
		jump = 0;
		player.jump(JUMP_FORCE);
	}
	
	if (jump < 2 && !player.grounded()) {
		player.jump(JUMP_FORCE);
		jump = 1 ;
		
	}
	if (jump == 1 && !player.grounded()) {
		player.grounded();
		jump = 0;
			}
	
	
});`

**Comment by slmjkdbtl** [comment]
there's double jump in the ["sticky"](https://kaboomjs.com/examples#sticky) example where it's packed into a component (maybe i should just renamed it to "doublejump"):

```js
// custom component to handle double jump
function djump() {
	let hasDouble = true;
	return {
		id: "djump",
		// requires the "body" component
		require: [ "body", ],
		// runs once on add
		add() {
			this.on("grounded", () => {
				hasDouble = true;
			});
		},
		djump(...args) {
			if (this.grounded()) {
				this.jump(...args);
			} else if (hasDouble) {
				hasDouble = false;
				this.jump(...args);
				// triggers a custom event
				this.trigger("djump");
			}
		},
	};
}

// add player character
const player = add([
	sprite("car"),
	area(),
	body(),
	pos(),
	// use our custom component here
	djump(),
]);

// use the patched djump() instead of normal jump()
keyPress("space", () => {
	player.djump();
});
```
